lib/SieveKeywordRegistry.php

```diff
@@ -80,8 +83,44 @@ public function activate($extension)
             return;
         }
 
+        if (in_array($extension, $this->requiredExtensions)) {
+            // already loaded
+            return;
+        } else {
+            $this->requiredExtensions[] = $extension;
```

Comment:
What do you think about making this a map?
`$this->requiredExtensions[$extension] = true`